flink/src/test/java/org/apache/iceberg/flink/TestFlinkSchemaUtil.java

```diff
@@ -218,6 +218,24 @@ public void testListField() {
             ))
     );
 
-    Assert.assertEquals(expectedSchema.asStruct(), actualSchema.asStruct());
+    checkSchema(flinkSchema, icebergSchema);
+  }
+
+  private void checkSchema(TableSchema flinkSchema, Schema icebergSchema) {
+    Assert.assertEquals(icebergSchema.asStruct(), FlinkSchemaUtil.convert(flinkSchema).asStruct());
+    // The conversion is not a 1:1 mapping, so we just check iceberg types.
+    Assert.assertEquals(
+        icebergSchema.asStruct(),
+        FlinkSchemaUtil.convert(FlinkSchemaUtil.toSchema(FlinkSchemaUtil.convert(icebergSchema))).asStruct());
```

Comment:
I'd prefer to provide some separate unit tests to address the conversion differences ( so that we won't regress).

Reply:
I'll add tests for the conversion differences.